app/models/bike.rb

```diff
@@ -625,6 +627,55 @@ def registration_location
     [city, state].reject(&:blank?).join(", ")
   end
 
+  def location_info_present?(record)
+    return false if record.blank?
+
+    if record.respond_to?(:country)
+      record.country.present? &&
+        (record.city.present? || record.zipcode.present?)
+    elsif record.respond_to?(:country_code)
+      record.country_code.present? &&
+        (record.city.present? || record.zipcode.present?)
+    end
+  end
+
+  # Set the bike's location (city, postal code, and country)
+  # based in the following order of precedence:
+  # 1. Set explicitly on the bike
+  # 2. From the current stolen record, if one is present
+  # 3. From the creation organization, if one is present
+  # 4. From the bike owner's address
+  # 5. From the request's IP address
+  def set_location_info(request_location: nil)
```

Comment:
I'm not sure this is currently the case, and if it is I think it's something we should solve in a future PR - but I want to note it now.

Something I just realized we ought to do is make it so if a user removes the location from their bike, we respect that and don't set the location via other methods.